### PR TITLE
Update logger to facilitate the use of global logger

### PR DESCRIPTION
**Pull request by rawmind0**
This PR add the chance to use global zap logger within the packages. Also some code cleanup has been done, like remove the logger mutex due to zap package already take care of them

:link: [zboto Link](https://app.clickup.com/t/8696dxv0b)

### CODE REVIEW

**Review by emmanuelm41 — APPROVED** [review]
I really like the improvement. I would improve a bit the readme though. Maybe using chat gpt, providing code as input?

File: pkg/cli/root.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ type AppSettings struct {
 	EnvPrefix   string // environment variable MYAPP_.....
 	GitVersion  string
 	GitRevision string
+	LogLevel    string // Global log level for the app
```
Comment (jleni) [inline pkg/cli/root.go:19]: If we add this. Do we have a backwards compatible path?
Are missing params in the yaml acceptable or it will throw an error?

Comment (rawmind0) [reply pkg/cli/root.go:19]: The new parameter is optional. By default the logger is initialized using info level and json encode. Just if LogLevel is defined then the logger is updated, https://github.com/Zondax/golem/pull/150/files#diff-fa1c0f512f57498ed99a3473226bacfba1be2c794c4a1ff08758fd53c9a338b8R72

File: pkg/runner/runner.go [hunk 2]
```diff
@@ -61,7 +63,7 @@ func (tr *TaskRunner) sendError(te TaskError) {
 		break
 	default:
 		// FIXME: log error at least? Nobody is registered to receive errors
```
[the remark this answers is not in the document]
Comment (rawmind0) [reply pkg/runner/runner.go:65]: Oh yeah, i updated the fmt to log but it missed to remove the comment.

**Comment by rawmind0** [comment]
> I really like the improvement. I would improve a bit the readme though. Maybe using chat gpt, providing code as input?

Sure, we may try this with cursor?? would be pretty nice. I've just added a pretty basic README

File: pkg/cli/root.go (hunk 1 above)
Comment (emmanuelm41) [reply pkg/cli/root.go:19]: Nice

**Comment by emmanuelm41** [comment]
@rawmind0 yes, using cursor is what I meant!!

**Comment by rawmind0** [comment]
Ok, let me check if i can do it.

**Comment by rawmind0** [comment]
Logger README has been updated using ia

File: pkg/logger/README.md [hunk 3]
```diff
@@ -1,40 +1,174 @@
...
+- Context-aware logging
+- Global and instance-based logging
+- Configurable log levels and encoding formats
+- Request ID tracking support
+- Easy integration with existing applications
+
+## Installation
+
+```go
+go get -u go.uber.org/zap
```
Comment (emmanuelm41) [inline pkg/logger/README.md:18]: go get zap? I don't think this is correct!

Comment (rawmind0) [reply pkg/logger/README.md]: You are right.  Fixed

**Pull request merged**